app/graphql/types/query_type.rb

```diff
@@ -0,0 +1,89 @@
+Types::QueryType = GraphQL::ObjectType.define do
+  name 'Query'
+
+  field :user do
+    type Types::UserType
+    argument :id, !types.ID
+    description 'Find a User by ID'
+    resolve ->(obj, args, ctx) do
+      object = User.find(args['id'])
+      UserPolicy.new(ctx[:current_user], object).show? ? (object) : nil
+    end
+  end
+
+  field :users, types[Types::UserType] do
+    argument :limit, types.Int, default_value: 20
+    resolve ->(obj, args, ctx) do
+      limit = [args[:limit], 30].min
```

Comment:
Everything is fine, the argument is not required to specify, so I set the default value, and I do the query with an argument to show how it works with an argument other than the default value